### PR TITLE
Use 're' for all RecEditor in RecCard

### PR DESCRIPTION
**Pull request by cydanil**
RecCard would have either `recipe_editor` or `re`, depending how it was instantiated.

This was [spotted by @takluyver in #77](https://github.com/kirienko/gourmet/pull/77#discussion_r443119323). That PR will have to rebase and the tests be updated accordingly.

I also removed an obsolete TODO comment, which was discussed in #72.

### CODE REVIEW

**Comment by cydanil** [comment]
The changes were originally in `RecCard.show_edit()`, other changes around it followed.

**Comment by saxon-s** [comment]
Is it just me ... I tend to associate `re` with regex (i.e. `re.py`) and not `recipe_editor`.

**Comment by cydanil** [comment]
I also agree that `recipe_editor` is more telling. However, I find that `re` is more in the style of the application: it was more present than `recipe_editor`, and keeps consistency with `rc`, `rd`, `rg`.

**Comment by maweki** [comment]
re is very confusing, especially since that's the regex library. I'm all in favor of long names. I rather we don't have complicated logic where we need a lot of those names in one line than short names with many of them in one expression.

**Comment by takluyver** [comment]
I agree that abbreviations are in the style of the existing code, but I'm not sure that style is particularly good, and like the others, `re` makes me think regex.

I think it's reasonable to abbreviate 'recipe' in a recipe manager program, though, so I'd be happy with an abbreviation like `rec_edit` or `r_editor`.

**Comment by cydanil** [comment]
Ok, I settled on `rec_editor` since it's also the name of the class :)

File: gourmet/reccard.py [hunk 1]
```diff
@@ -120,8 +121,9 @@ def update_recipe (self, recipe):
         self.current_rec = recipe
         if hasattr(self,'recipe_display'):
             self.recipe_display.update_from_database()
-        if self.re is not None and not self.re.window.is_visible():
-            self.re = None
+        if (self.rec_editor is not None and
+            not self.rec_editor.window.is_visible()):
```
Comment (takluyver) [inline gourmet/reccard.py:125]: I'm open to doing checks like this as `self.rec_editor and self.rec_editor.window.is_visible()`, especially if it means that it fits on one line. But I'm not going to push for that if you prefer the specific `is not None` check.

**Comment by takluyver** [comment]
LGTM

File: gourmet/reccard.py (hunk 1 above)
Comment (cydanil) [reply gourmet/reccard.py:125]: Good idea, I was trying to keep everything clear in my mind at the time.

Comment (maweki) [reply gourmet/reccard.py:125]: The `is not None` check is specifically supported by `mypy` as means to resolve `Optional[A]` into `A`. I am all for it.

Comment (cydanil) [reply gourmet/reccard.py:125]: Okay, let's keep it :)

**Comment by cydanil** [comment]
Thanks for the reviews! Stay tuned for more of the same

**Pull request merged**